server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadata.java

```diff
@@ -167,165 +157,18 @@ protected NodeStoreFilesMetadata nodeOperation(NodeRequest request) {
-
-        public StoreFilesMetadata(
-            ShardId shardId,
-            Store.MetadataSnapshot metadataSnapshot,
-            List<RetentionLease> peerRecoveryRetentionLeases
-        ) {
-            this.shardId = shardId;
-            this.metadataSnapshot = metadataSnapshot;
-            this.peerRecoveryRetentionLeases = peerRecoveryRetentionLeases;
-        }
-
-        public StoreFilesMetadata(StreamInput in) throws IOException {
-            this.shardId = new ShardId(in);
-            this.metadataSnapshot = new Store.MetadataSnapshot(in);
-            this.peerRecoveryRetentionLeases = in.readList(RetentionLease::new);
-        }
-
-        @Override
-        public void writeTo(StreamOutput out) throws IOException {
-            shardId.writeTo(out);
-            metadataSnapshot.writeTo(out);
-            out.writeList(peerRecoveryRetentionLeases);
-        }
-
-        public ShardId shardId() {
-            return this.shardId;
-        }
-
-        public boolean isEmpty() {
-            return metadataSnapshot.size() == 0;
-        }
-
-        @Override
-        public Iterator<StoreFileMetadata> iterator() {
-            return metadataSnapshot.iterator();
-        }
-
-        public boolean fileExists(String name) {
-            return metadataSnapshot.asMap().containsKey(name);
-        }
-
-        public StoreFileMetadata file(String name) {
-            return metadataSnapshot.asMap().get(name);
-        }
-
-        /**
-         * Returns the retaining sequence number of the peer recovery retention lease for a given node if exists; otherwise, returns -1.
-         */
-        public long getPeerRecoveryRetentionLeaseRetainingSeqNo(DiscoveryNode node) {
-            assert node != null;
-            final String retentionLeaseId = ReplicationTracker.getPeerRecoveryRetentionLeaseId(node.getId());
-            return peerRecoveryRetentionLeases.stream()
-                .filter(lease -> lease.id().equals(retentionLeaseId))
-                .mapToLong(RetentionLease::retainingSequenceNumber)
-                .findFirst()
-                .orElse(-1L);
-        }
-
-        public List<RetentionLease> peerRecoveryRetentionLeases() {
-            return peerRecoveryRetentionLeases;
-        }
-
-        /**
-         * @return commit sync id if exists, else null
-         */
-        public String syncId() {
-            return metadataSnapshot.getSyncId();
-        }
-
-        @Override
-        public String toString() {
-            return "StoreFilesMetadata{"
-                + ", shardId="
-                + shardId
-                + ", metadataSnapshot{size="
-                + metadataSnapshot.size()
-                + ", syncId="
-                + metadataSnapshot.getSyncId()
-                + "}"
-                + '}';
+        } catch (IOException e) {
+            return new StoreFilesMetadata(shardId, Store.MetadataSnapshot.EMPTY, Collections.emptyList());
```

Comment:
@shiv0408 to maintain the old transport as it is, let's throw exception from here as it was being done. 
We only need modification(sending exception as part of response) in the new transport.

Reply:
We were not throwing exception before as well in case of IOException or IndexNotFoundException (which is anyway a subclass of IOExcpetion), so the behavior is not changed

https://github.com/opensearch-project/OpenSearch/blob/76ae14a4f2e99592610d3181543f5036214ceb7a/server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadata.java#L186-L191

Reply:
@shwetathareja IndexNotFoundException is an extension of IOException. Therefore the above code, where we are just catching IOException will be sufficient to catch and return empty response in case of IndexNotFoundException.